run-tests.php

```diff
@@ -3720,9 +3720,14 @@ public function checkSkip(string $php, string $code, string $checkFile, string $
 
         save_text($checkFile, $code, $tempFile);
         $result = trim(system_with_timeout("$php \"$checkFile\"", $env));
-        if (strpos($result, 'nocache') === 0) {
-            $result = '';
-        } else if ($this->enable) {
+        $cacheResult = $this->enable;
+
+        while (!strncasecmp('nocache', $result, 7)) {
+            $result = ltrim(substr($result, 7));
+            $cacheResult = false;
+        }
+
+        if ($cacheResult) {
```

Comment:
> I'm slightly concerned about this if we actually target PHP-8.1.

@cmb69: This change should address your previous concern. One or more "nocache" tags will now be consumed exhaustively, which will avoid leading behind a "nocache" that might otherwise cause the SKIPIF processing in `run_test()` to BORK.

This also preserves the a ability to "skip a test _and_ disable caching" by emitting "nocache" at the beginning of SKIPIF output, which I mentioned in the PR OP.